src/main/java/io/permit/sdk/enforcement/Enforcer.java

```diff
@@ -448,6 +449,63 @@ public UserPermissions getUserPermissions(GetUserPermissionsQuery input) throws
         return result;
     }
 
+
+    @Override
+    public UserPermissions getUserPermissionsWithOPA(GetUserPermissionsQuery input) throws IOException, PermitApiError {
+        // request body
+        Gson gson = new Gson();
+
+         // Inner map for the nested JSON
+        Map<String, Object> innerMap = new HashMap<>();
+        innerMap.put("user", input.user);
+        innerMap.put("tenants", input.tenants);
+        innerMap.put("resource_types", input.resource_types);
+        innerMap.put("resources", input.resources);
+        innerMap.put("context", input.context);
+
+        // Outer map wrapping the inner map
+        Map<String, Object> outerMap = new HashMap<>();
+        outerMap.put("input", innerMap);
+
+        // Serialize to JSON
+        String requestBody = gson.toJson(outerMap);
```

Comment:
is this actually the fastest way to do that ?

Reply:
i don't know java - i check again

Reply:
There is a way - but we are using java8 which not support that functionality 
```
import java.util.HashMap;
import java.util.Map;

public class JsonBuilder {
    public static String buildRequestBody(Input input) {
        Gson gson = new Gson();

        Map<String, Object> innerMap = new HashMap<>();
        innerMap.put("user", input.user);
        innerMap.put("tenants", input.tenants);
        innerMap.put("resource_types", input.resource_types);
        innerMap.put("resources", input.resources);
        innerMap.put("context", input.context);

        Map<String, Object> outerMap = new HashMap<>();
        outerMap.put("input", innerMap);

        return gson.toJson(outerMap);
    }
}
```